src/main/java/mods/railcraft/common/plugins/forge/WorldPlugin.java

```diff
@@ -140,12 +140,11 @@ public static boolean playerRemoveBlock(World world, BlockPos pos, @Nullable Ent
         if (MinecraftForge.EVENT_BUS.post(new BlockEvent.BreakEvent(world, pos, state, player)))
             return false;
 
-        if (state.getBlock().removedByPlayer(state, world, pos, player, dropBlock))
+        if (!state.getBlock().removedByPlayer(state, world, pos, player, dropBlock))
             return false;
 
         if (dropBlock) {
-            List<ItemStack> drops = state.getBlock().getDrops(world, pos, state, 0);
-            InvTools.dropItems(drops, world, pos);
+            state.getBlock().harvestBlock(world, player, pos, state, getBlockTile(world, pos), player.getHeldItem(player.getActiveHand()));
```

Comment:
That simply says if you want to keep the tile you can keep it in destroyBlock and remove it in harvestBlock, or remove it directly when it does not drop or when the drop does not depend on the tile entity. So my code is fine.